executor/src/witgen/machines/fixed_lookup_machine.rs

```diff
@@ -279,14 +284,43 @@ impl<T: FieldElement> FixedLookup<T> {
 
     fn process_range_check<'b>(
         &self,
+        rows: &RowPair<'_, '_, T>,
         lhs: &AffineExpression<&'b PolynomialReference, T>,
         rhs: &'b PolynomialReference,
     ) -> EvalResult<'b, T> {
+        // Use AffineExpression::solve_with_range_constraints to transfer range constraints
+        // from the rhs to the lhs.
         let equation = lhs.clone() - AffineExpression::from_variable_id(rhs);
-        // TODO this is kind of unusual because we use symbolic fixed columns.
-        // we should better filter out the fixed column from the result.
-        let r = equation.solve_with_range_constraints(&self.global_constraints);
-        println!("{r:?}");
-        r
+        let range_constraints = UnifiedRangeConstraints {
+            witness_constraints: rows,
+            global_constraints: &self.global_constraints,
+        };
+        let updates = equation.solve_with_range_constraints(&range_constraints)?;
+
+        // Filter out any updates to the fixed columns and mark the identity as complete
```

Comment:
Oh right, it is complete! Although we should maybe only do that if the RHS is "full", i.e. has all the values of the range constraint.

Reply:
... and if we actually know a range constraint on the RHS at all.

Reply:
Also, what if we have a lookup of the form `{x + y} in {z}`, and we don't know `x` nor `y`. But the later on we learn `y = 0`. Then we should run the lookup again, shouldn't we?

Reply:
Hmm, good point! I changed it to incomplete. This code cannot set a concrete value, so it will always be incomplete. If the LHS already has a concrete value, this code won't be executed.